packages/lodestar/src/network/util.ts

```diff
@@ -138,24 +138,32 @@ export async function dialProtocol(
   signal?: AbortSignal
 ): ReturnType<LibP2p["dialProtocol"]> {
   const abortController = new AbortController();
-  const timer = setTimeout(() => {
-    abortController.abort();
-  }, timeout);
   const abortHandler = (): void => abortController.abort();
   signal?.addEventListener("abort", abortHandler, {once: true});
+  let timer: NodeJS.Timeout | null = null;
   try {
-    const conn = await libp2p.dialProtocol(peerId, protocol, {signal: abortController.signal} as object);
+    const conn = await Promise.race([
```

Comment:
Is it worth to abstract this construction so we can properly test it and make sure it doesn't leak?
```ts
abortableTimeoutCallback(
  fn: () => Promise<void>,
  timeoutMs: number,
  signal: AbortSignal
)
```

Reply:
Yeah, I think that would be really useful to pull into its own function. Basically like `setTimeout` but with a signal. (and not supporting the 'evil' extra arguments that `setTimeout` supports)